### ISSUE TITLE
Delete valet.sock on `valet start`, `valet stop`, and `valet restart`

### ISSUE THREAD

**Issue by mattstauffer** [laravel/valet#1102]
## Clear description of your problem

In many circumstances `~/.config/valet/valet.sock` still persists when it shouldn't, leaving me with the wrong version of Valet running.

## Expected behavior

If Valet is running a version of PHP locally, I expect that same version to be what's running on web sites that are being served.

## Current behavior

Often I'll find that, when I restart my computer, or in other scenarios--still trying to exactly pin it down--I'll have PHP 8 running on the command line, managed by Valet, but a site will open and give a Composer error because I'm running PHP 7.1 or something.

## Steps to Reproduce

1. I still can't figure out exactly how to repro or what the source of this is, frustratingly!!

## Possible solution

Every time we run `valet stop` or `valet start` or `valet restart`, delete the `valet.sock` file if it exists.

We already do this when running `valet use`.

## Diagnosis

Not gonna run the whole thing, but I'm on v2.15.0 on one machine and v2.16.0 on the other and it's happening on both.

**Comment by nicoverbruggen** [comment]
Heya @mattstauffer and @drbyte, I think I have some (hopefully) helpful info related to this particular issue.

I’m currently preparing for the release of [phpmon 4.1](https://github.com/nicoverbruggen/phpmon/tree/develop), which will use Valet’s own `valet use` command to switch between different versions of PHP! This was primarily done to avoid potential issues caused by the app invoking the `brew link` and `brew services` commands on its own.

I've been doing some testing over the past week and I bumped into an issue today after rebooting my machine, and I think it's a potential bug in Valet. (I was going to make a new issue but this particular issue *is the very thing I'm bumping into as well*.)

On my Mac, I have the latest version of Valet running and the `php` and `php@8.0` formulae (from Homebrew Core) are installed.

I have noticed that sometimes, whenever I reboot my machine, sites are being *served* using PHP 8.1, even though I switched to PHP 8.0 before rebooting. It is important to note that PHP 8.0 remains linked correctly after the reboot. 

(Here’s the wonky part: I cannot reproduce this *every* time. Switching versions and rebooting does not always seem to trigger this particular behaviour, so I'm a bit puzzled. I suspect this is a timing issue...)

After switching to `php@8.0` (using `valet use php@8.0` in the terminal) I went to inspect the output of `sudo brew services list`, and I noticed the following :

```
➜  ~ sudo brew services list
Name    Status  User File
dnsmasq started root /Library/LaunchDaemons/homebrew.mxcl.dnsmasq.plist
httpd   none
mailhog none
nginx   started root /Library/LaunchDaemons/homebrew.mxcl.nginx.plist
php     started root /Library/LaunchDaemons/homebrew.mxcl.php.plist
php@8.0 started root /Library/LaunchDaemons/homebrew.mxcl.php@8.0.plist
unbound none
```

Here's what surprised me: both the `php` *and* `php@8.0` FPM services seem to be running... 

The way I understand it, the following happens: because the (not actively linked) `php` formula's related FPM service is still active (even after switching to `php@8.0`) the LaunchDaemon file still exists, so after a reboot my sites are being (incorrectly) served using PHP 8.1, all while the PHP 8.0 binary remains (correctly) linked.

This happens because the PHP-FPM service of PHP 8.1 boots up when the system reboots. If the `php` service (which corresponds to PHP 8.1) had been disabled after running `valet use php@8.0` this would not have been an issue, I believe.

I'm not familiar with Valet's codebase it its entirely, but I would expect the `php` service (or any other `php@X.X` aliases *except* for the desired version) to be disabled once one attempts to switch to another version of PHP using `valet use php@{version}`. 

I believe running those multiple PHP-FPM processes upon rebooting might be the reason the socket file is incorrectly created as well.

When I manually stop the `php` service (`brew services stop php`) and reboot my computer, the sites always seem to be served using the correct PHP version. So if Valet were to stop all `php` services not required, this could potentially solve that issue?

It is still the case that Valet only serves *one* particular version of PHP at the time, right? No need to have those extra PHP-FPM processes running then, I'd imagine. If you have many versions of PHP installed you could potentially have 5+ interfering PHP-FPM processes up and running... most of which are not going to be used by nginx and Valet, right?

Hope this helps somehow!

**Comment by mattstauffer** [comment]
Great research on this @nicoverbruggen!! Taking a look at your PR now :)

**Comment by driesvints** [comment]
@mattstauffer going to close this one as it seems it's resolved with #1178
